### ISSUE TITLE
An empty profile returns an un-useful error

### ISSUE THREAD

**Issue by clrcrl** [fishtown-analytics/dbt#2292]
### Steps To Reproduce
1. Create a profile that's empty (in `profiles.yml`):
```
jaffle_shop: 
```
2. Ensure your project uses this profile
2. Run any dbt command
### Expected behavior
Some sort of helpful error message, like "your profile is empty"

### Actual behaviour
```
Running with dbt=0.16.0
Encountered an error:
'NoneType' object has no attribute 'get'
```
### Screenshots and log output
If applicable, add screenshots or log output to help explain your problem.

### System information

**The output of `dbt --version`:**
0.16.0

**Comment by drewbanin** [comment]
Ooh, good catch, thanks @clrcrl! This was actually fixed (maybe inadvertently) in https://github.com/fishtown-analytics/dbt/pull/2290

```
09:51:26 ~/fishtown/clients/debug $ head dbt_project.yml

name: 'my_project'
version: '1.0'

profile:

09:51:30 ~/fishtown/clients/debug $ dbt run
Running with dbt=0.16.1-a1
Encountered an error while reading the project:
  ERROR: Runtime Error
  dbt cannot run because no profile was specified for this dbt project.
  To specify a profile for this project, add a line like the this to
  your dbt_project.yml file:

  profile: [profile name]

  Here, [profile name] should be replaced with a profile name
  defined in your profiles.yml file. You can find profiles.yml here:

  /Users/drew/.dbt/profiles.yml

Encountered an error:
Runtime Error
  Could not run dbt
```


It's going to ship in 0.16.1 next week :)

**Comment by clrcrl** [comment]
Hey @drewbanin, I'm not sure if that's the same issue.

I'm talking about having an empty profile in your `profiles.yml`, whereas the above is talking about having an empty `profile:` key in your `dbt_project.yml`

**Comment by drewbanin** [comment]
oooh, i understand, thanks for clarifying!

Adding this to the Octavius Catto milestone, should be a quick fix

**Comment by mshayeb** [comment]
I got error when I tried to run: dbt run:

# dbt_project.yml

name: 'jaffle_shop'
version: '0.1'
profile: 'my_dbt'
source-paths: ["models"]
analysis-paths: ["analysis"]
test-paths: ["tests"]
data-paths: ["data"]
macro-paths: ["macros"]

target-path: "target"
clean-targets:
    - "target"
    - "dbt_modules"
    - "logs"

models:
  jaffle_shop:
      materialized: table
      staging:
        materialized: view
        tags: ["staging", "hourly"]



Running with dbt=0.16.1
Encountered an error:
'NoneType' object has no attribute 'get'

**Comment by drewbanin** [comment]
hey @mshayeb - this is fixed in the _next_ release of dbt, v0.17.0, due out around mid-May:

https://github.com/fishtown-analytics/dbt/pull/2297
